pkg/operation/botanist/kubeapiserver.go

```diff
@@ -156,6 +191,72 @@ func (b *Botanist) computeKubeAPIServerAdmissionPlugins(defaultPlugins, configur
 	return admissionPlugins
 }
 
+func (b *Botanist) ensureAdmissionPluginConfig(plugins []gardencorev1beta1.AdmissionPlugin, kubernetesVersion string) ([]gardencorev1beta1.AdmissionPlugin, error) {
+	var (
+		index                     int
+		allowPrivilegedContainers = pointer.BoolDeref(b.Shoot.GetInfo().Spec.Kubernetes.AllowPrivilegedContainers, false)
+		pspDisabled               = b.Shoot.PSPDisabled
+	)
+
+	for i, plugin := range plugins {
+		if plugin.Name == "PodSecurity" {
+			index = i
+		}
+	}
+
+	// if a config is set in the shoot spec
+	if plugins[index].Config != nil {
+		config, err := runtime.Decode(codec, plugins[index].Config.Raw)
+		if err != nil {
+			return plugins, err
```

Comment:
Typically we return `nil` when an error occurs

```suggestion
			return nil, err
```

(same below)